diff --git a/.config-win32 b/.config-win32
new file mode 100644
index 0000000..e0e7b24
--- /dev/null
+++ b/.config-win32
@@ -0,0 +1,38 @@
+CONFIG_WINDOWS = 1
+CONFIG_32BIT = 1
+
+TARGET = gamex86.dll
+
+CC = i686-w64-mingw32-gcc
+LD = i686-w64-mingw32-gcc
+WINDRES = i686-w64-mingw32-windres
+
+CFLAGS = -std=gnu99 -static -static-libgcc -DCURL_STATICLIB -DHAVE_CURL -DWIN32 -D__USE_W32_SOCKETS
+LDFLAGS = -shared -static -static-libgcc
+
+LIBS = -L/usr/i686-w64-mingw32/lib
+
+# curl and its deps (only ones we know exist)
+LIBS += /usr/i686-w64-mingw32/lib/libcurl.a
+LIBS += /usr/i686-w64-mingw32/lib/libnghttp2.a
+LIBS += /usr/i686-w64-mingw32/lib/libpsl.a
+LIBS += /usr/i686-w64-mingw32/lib/libidn2.a
+LIBS += /usr/i686-w64-mingw32/lib/libbrotlidec.a
+LIBS += /usr/i686-w64-mingw32/lib/libbrotlicommon.a
+#LIBS += /usr/i686-w64-mingw32/lib/libzstd.dll.a
+LIBS += /usr/i686-w64-mingw32/lib/libunistring.a
+
+# ssh2 (needs zlib)
+LIBS += /usr/i686-w64-mingw32/lib/libssh2.a
+LIBS += /usr/i686-w64-mingw32/lib/libz.a
+
+# OpenSSL (ssl first, then crypto)
+LIBS += /usr/i686-w64-mingw32/lib/libssl.a
+LIBS += /usr/i686-w64-mingw32/lib/libcrypto.a
+
+# windows / misc helpers
+LIBS += /usr/i686-w64-mingw32/lib/libcrypt32.a
+LIBS += /usr/i686-w64-mingw32/lib/libwldap32.a
+LIBS += /usr/i686-w64-mingw32/lib/libiconv.a
+LIBS += /usr/i686-w64-mingw32/lib/libbcrypt.a
+ 
diff --git a/.config-win64 b/.config-win64
new file mode 100644
index 0000000..c77896b
--- /dev/null
+++ b/.config-win64
@@ -0,0 +1,37 @@
+CONFIG_WINDOWS = 1
+
+TARGET = gamex86_64.dll
+
+CC = x86_64-w64-mingw32-gcc
+LD = x86_64-w64-mingw32-gcc
+WINDRES = x86_64-w64-mingw32-windres
+
+CFLAGS = -std=gnu99 -static -static-libgcc -DCURL_STATICLIB -DHAVE_CURL -DWIN32 -D__USE_W32_SOCKETS
+LDFLAGS = -shared -static -static-libgcc
+
+LIBS = -L/usr/x86_64-w64-mingw32/lib
+
+# curl and its deps (only ones we know exist)
+LIBS += /usr/x86_64-w64-mingw32/lib/libcurl.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libnghttp2.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libpsl.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libidn2.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libbrotlidec.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libbrotlicommon.a
+#LIBS += /usr/x86_64-w64-mingw32/lib/libzstd.dll.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libunistring.a
+
+# ssh2 (needs zlib)
+LIBS += /usr/x86_64-w64-mingw32/lib/libssh2.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libz.a
+
+# OpenSSL (ssl first, then crypto)
+LIBS += /usr/x86_64-w64-mingw32/lib/libssl.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libcrypto.a
+
+# windows / misc helpers
+LIBS += /usr/x86_64-w64-mingw32/lib/libcrypt32.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libwldap32.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libiconv.a
+LIBS += /usr/x86_64-w64-mingw32/lib/libbcrypt.a
+
diff --git a/.gitignore b/.gitignore
index 6f98af9..21fdd13 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,3 @@
-/.config*
 /*.d
 /*.o
 /game*.so
diff --git a/Makefile b/Makefile
index 03aba4b..ff2bbf1 100644
--- a/Makefile
+++ b/Makefile
@@ -1,19 +1,17 @@
-### OpenFFA Makefile ###
-
-CONFIG_FILE ?= .config
-
--include $(CONFIG_FILE)
+ifdef CONFIG_FILE
+	-include $(CONFIG_FILE)
+endif
 
 ifndef CPU
-    CPU := $(shell uname -m | sed -e s/i.86/i386/ -e s/amd64/x86_64/ -e s/sun4u/sparc64/ -e s/arm.*/arm/ -e s/sa110/arm/ -e s/alpha/axp/)
+	CPU := $(shell uname -m | sed -e s/i.86/i386/ -e s/amd64/x86_64/ -e s/sun4u/sparc64/ -e s/arm.*/arm/ -e s/sa110/arm/ -e s/alpha/axp/)
 endif
 
 ifndef REV
-    REV := $(shell git rev-list --count HEAD)
+	REV := $(shell git rev-list --count HEAD)
 endif
 
 ifndef VER
-    VER := r$(REV)~$(shell git rev-parse --short HEAD)
+	VER := r$(REV)~$(shell git rev-parse --short HEAD)
 endif
 
 CC ?= gcc
@@ -73,6 +71,14 @@ ifdef CONFIG_UDP
     OBJS += g_udp.o
 endif
 
+ifdef CONFIG_32BIT
+	CFLAGS += -m32
+	LDFLAGS += -m32
+	CPU = x86
+else
+	CPU = x86_64
+endif
+
 ifdef CONFIG_WINDOWS
     OBJS += openffa.o
     LIBS += -lws2_32
diff --git a/README-MashedD.md b/README-MashedD.md
new file mode 100644
index 0000000..0fc92c6
--- /dev/null
+++ b/README-MashedD.md
@@ -0,0 +1,57 @@
+# openffa
+
+# Notes/commands
+
+-
+
+# Building
+
+Tested on CachyOS.
+
+## Prerequisites
+
+Dependencies might be missing and some are Probably excessive.
+I didn't optimize this as it's time/cost not effective for me.
+Best might be to use Docker for the job.
+
+```bash
+sudo pacman -S cmake gcc
+
+# For cross compilation
+sudo pacman -S \
+    mingw-w64-tools \
+    mingw-w64-binutils \
+    mingw-w64-crt \
+    mingw-w64-gcc \
+    mingw-w64-headers \
+    mingw-w64-winpthreads
+paru -S \
+    mingw-w64-zlib \
+    mingw-w64-zlib-ng \
+    mingw-w64-ffmpeg \
+    mingw-w64-pkg-config \
+    mingw-w64-libpng \
+    mingw-w64-libjpeg-turbo \
+    mingw-w64-openal \
+    mingw-w64-zstd
+```
+
+## Compilation
+
+Review scripts before executing them.
+
+```bash
+./build-lin64.sh
+./clean.sh
+./build-win32.sh
+./clean.sh
+./build-win64.sh
+```
+
+# TODO
+
+- fork latest repo
+- apply changes in separate commits
+- add/change README.md
+- fix compilation warnings
+
diff --git a/build-lin64.sh b/build-lin64.sh
new file mode 100755
index 0000000..137be4a
--- /dev/null
+++ b/build-lin64.sh
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+#make clean
+make -j$(nproc)
+
+DEST="../../openffa"
+if [ -e "$DEST" ]; then
+    cp -f gamex86_64.so "$DEST/"
+    #cd "$DEST"
+    #./start-q2pro-lin64-mashedd.sh
+fi
+
diff --git a/build-win32.sh b/build-win32.sh
new file mode 100755
index 0000000..bde02b3
--- /dev/null
+++ b/build-win32.sh
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+#make CONFIG_FILE=.config-win32 clean
+make CONFIG_FILE=.config-win32 -j$(nproc)
+chmod -x gamex86.dll
+
+DEST="../../openffa"
+if [ -e "$DEST" ]; then
+    cp -f gamex86.dll "$DEST/"
+    #cd "$DEST"
+    #./start-q2pro-win32-mashedd.sh
+fi
+
diff --git a/build-win64.sh b/build-win64.sh
new file mode 100755
index 0000000..eb8db3e
--- /dev/null
+++ b/build-win64.sh
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+#make CONFIG_FILE=.config-win64 clean
+make CONFIG_FILE=.config-win64 -j$(nproc)
+chmod -x gamex86_64.dll
+
+DEST="../../openffa"
+if [ -e "$DEST" ]; then
+    cp -f gamex86_64.dll "$DEST/"
+    #cd "$DEST"
+    #./start-q2pro-win64-mashedd.sh
+fi
+
diff --git a/clean.sh b/clean.sh
new file mode 100755
index 0000000..fc92909
--- /dev/null
+++ b/clean.sh
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+rm -rf debug
+rm -rf release
+make clean
+make CONFIG_FILE=.config-win32 clean
+make CONFIG_FILE=.config-win64 clean
+
